src/arm/radxa_cm3_io.c

```diff
@@ -0,0 +1,186 @@
+    b->pins[37].pwm.parent_id = 3;
+    b->pins[37].pwm.mux_total = 0;
+    b->pins[37].pwm.pinmap = 0;
+
+    b->aio_count = MRAA_RADXA_CM3_IO_AIO_COUNT;
+    b->adc_raw = 10;
+    b->adc_supported = 10;
+    b->aio_dev[0].pin = 26;
+    b->aio_non_seq = 1;
+
+    mraa_radxa_cm3_io_pininfo(b, 0,   -1, (mraa_pincapabilities_t){0,0,0,0,0,0,0,0}, "INVALID");
+    mraa_radxa_cm3_io_pininfo(b, 1,   -1, (mraa_pincapabilities_t){1,0,0,0,0,0,0,0}, "3V3");
+    mraa_radxa_cm3_io_pininfo(b, 2,   14, (mraa_pincapabilities_t){1,0,0,0,0,0,0,0}, "5V");
+    mraa_radxa_cm3_io_pininfo(b, 3,   -1, (mraa_pincapabilities_t){1,1,0,0,0,1,0,0}, "SDA2");
+    mraa_radxa_cm3_io_pininfo(b, 4,   -1, (mraa_pincapabilities_t){1,0,0,0,0,0,0,0}, "5V");
+    mraa_radxa_cm3_io_pininfo(b, 5,   13, (mraa_pincapabilities_t){1,1,0,0,0,1,0,0}, "SCL2");
+    mraa_radxa_cm3_io_pininfo(b, 6,   -1, (mraa_pincapabilities_t){1,0,0,0,0,0,0,0}, "GND");
+    mraa_radxa_cm3_io_pininfo(b, 7,  125, (mraa_pincapabilities_t){1,1,0,0,0,0,0,0}, "GPIO3_D5");
+    mraa_radxa_cm3_io_pininfo(b, 8,   25, (mraa_pincapabilities_t){1,1,0,0,0,0,0,1}, "TXD2");
+    mraa_radxa_cm3_io_pininfo(b, 9,   -1, (mraa_pincapabilities_t){1,0,0,0,0,0,0,0}, "GND");
+    mraa_radxa_cm3_io_pininfo(b, 10,  24, (mraa_pincapabilities_t){1,1,0,0,0,0,0,1}, "RXD2");
+    mraa_radxa_cm3_io_pininfo(b, 11,  23, (mraa_pincapabilities_t){1,1,1,0,0,0,0,0}, "PWM0_M0");
+    mraa_radxa_cm3_io_pininfo(b, 12, 119, (mraa_pincapabilities_t){1,1,0,0,0,0,0,0}, "GPIO3_A7");
+    mraa_radxa_cm3_io_pininfo(b, 13,  15, (mraa_pincapabilities_t){1,1,1,0,0,0,0,0}, "PWM0_M1");
+    mraa_radxa_cm3_io_pininfo(b, 14,  -1, (mraa_pincapabilities_t){1,0,0,0,0,0,0,0}, "GND");
+    mraa_radxa_cm3_io_pininfo(b, 15,  19, (mraa_pincapabilities_t){1,1,1,0,0,0,0,0}, "PWM4");
+    mraa_radxa_cm3_io_pininfo(b, 16, 124, (mraa_pincapabilities_t){1,1,0,0,0,0,0,0}, "GPIO3_D4");
+    mraa_radxa_cm3_io_pininfo(b, 17,  -1, (mraa_pincapabilities_t){1,0,0,0,0,0,0,0}, "3V3");
+    mraa_radxa_cm3_io_pininfo(b, 18, 123, (mraa_pincapabilities_t){1,1,0,0,0,0,0,0}, "GPIO3_D3");
+    mraa_radxa_cm3_io_pininfo(b, 19, 138, (mraa_pincapabilities_t){1,1,0,0,1,0,0,1}, "SPI3_MOSI_M0");
+    mraa_radxa_cm3_io_pininfo(b, 20,  -1, (mraa_pincapabilities_t){1,0,0,0,0,0,0,0}, "GND");
+    mraa_radxa_cm3_io_pininfo(b, 21, 136, (mraa_pincapabilities_t){1,1,0,0,1,0,0,1}, "SPI3_MISO_M0");
+    mraa_radxa_cm3_io_pininfo(b, 22, 118, (mraa_pincapabilities_t){1,1,0,0,0,0,0,0}, "GPIO3_C6");
+    mraa_radxa_cm3_io_pininfo(b, 23, 139, (mraa_pincapabilities_t){1,1,0,0,1,0,0,1}, "SPI1CLK_M0");
+    mraa_radxa_cm3_io_pininfo(b, 24, 134, (mraa_pincapabilities_t){1,1,0,0,1,0,0,0}, "SPI3_CS0_M0");
+    mraa_radxa_cm3_io_pininfo(b, 25,  -1, (mraa_pincapabilities_t){1,0,0,0,0,0,0,0}, "GND");
+    mraa_radxa_cm3_io_pininfo(b, 26,  -1, (mraa_pincapabilities_t){1,0,0,0,0,0,1,0}, "SARADC");
+    mraa_radxa_cm3_io_pininfo(b, 27,  140, (mraa_pincapabilities_t){1,1,0,0,0,1,0,0}, "SDA2_M1");
```

Comment:
This and the following 2 lines need to have 1 space removed before 140, so everything can line up.